src/monmove.cpp

```diff
@@ -344,6 +365,22 @@ void monster::plan( const mfactions &factions )
             unset_dest();
         }
     }
+
+    // If we have no target, we're not hostile towards anyone, and not fleeing...
+    if( ( friendly >= 0 || target == nullptr ) && !fleeing ) {
+        // If we're a food-stealer, try to steal food, or possibly eat it if we have it
+        if( steals_food && wants_to_steal ) {
+            std::pair<item *, tripoint> item_target;
+            item_target = get_most_desired_visible_item( [&]( const item & itm ) {
```

Comment:
It would be cleaner to assign immediately: `const auto item_target = get_most_desired_visible_item(...`